src/tx/builder/field-types.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  writeId, readId, isNameValid, produceNameId, ensureNameValid, getMinimumNameFee, readInt, writeInt
+} from './helpers'
+import { InsufficientNameFeeError, IllegalArgumentError } from '../../utils/errors'
+import BigNumber from 'bignumber.js'
+
+/**
+ * Class implementing a common interface for all TX field types
+ * @class
+ */
+export class Field {
+  public prototype: any
+
+  /**
+   * Serialize value to Buffer
+   * @function
+   * @param {String | BigNumber} value
+   * @param {Object} _options
+   * @return {Buffer}
+   */
+  static serialize (value: string | BigNumber, _options: object = {}): Buffer {
+    return Buffer.from(value)
+  }
+
+  /**
+   * Deserialize value from Buffer
+   * @function
+   * @param {Buffer} value
+   * @return {String}
+   */
+  static deserialize (value: Buffer): string {
```

Comment:
```suggestion
  static deserialize (value: Buffer): any {
```
this is a general class and its inheritors can return actually anything, the same with arguments of `serialize`